src/action/macos/configure_remote_building.rs

```diff
@@ -0,0 +1,166 @@
+use crate::action::base::{create_or_insert_into_file, CreateOrInsertIntoFile};
+use crate::action::{
+    Action, ActionDescription, ActionError, ActionErrorKind, ActionTag, StatefulAction,
+};
+
+use std::path::Path;
+use tokio::task::JoinSet;
+use tracing::{span, Instrument, Span};
+
+const PROFILE_NIX_FILE_SHELL: &str = "/nix/var/nix/profiles/default/etc/profile.d/nix-daemon.sh";
+
+/**
+Configure macOS's zshenv to load the Nix environment when ForceCommand is used.
+This enables remote building, which requires `ssh host nix` to work.
+ */
+#[derive(Debug, serde::Deserialize, serde::Serialize, Clone)]
+pub struct ConfigureRemoteBuilding {
+    create_or_insert_into_files: Vec<StatefulAction<CreateOrInsertIntoFile>>,
+}
+
+impl ConfigureRemoteBuilding {
+    #[tracing::instrument(level = "debug", skip_all)]
+    pub async fn plan() -> Result<StatefulAction<Self>, ActionError> {
+        let mut create_or_insert_into_files = Vec::default();
+
+        let shell_buf = format!(
+            r#"
+# Nix, only on remote SSH connections -- for remote building.
+# See: <TICKET NUMBER>
```

Comment:
? :)

Reply:
Already fixed in my local copy, but haven't pushed it yet. Waiting for the CI run to complete so I can test it out on some hardware. But since lints failed, I'll push it now :).